yt_dlp/extractor/facebook.py

```diff
@@ -637,7 +638,7 @@ def parse_attachment(attachment, key='media'):
                 if len(entries) > 1:
                     return self.playlist_result(entries, video_id)
 
-                video_info = entries[0]
+                video_info = entries[0] if len(entries) > 0 else {'id': video_id}
```

Comment:
Could you explain this change?

Reply:
```suggestion
                video_info = entries[0] entries else {'id': video_id}
```

Reply:
without this change, "list index out of range" still occurs if line 613-615 cannot find anything.
with this change, "No format found" will be thrown.